server/src/apps/post/post.service.ts

```diff
@@ -1,17 +1,69 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadGatewayException,
+  BadRequestException,
+  ForbiddenException,
+  Injectable,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Posts } from 'src/entity/post.entity';
-import { Repository } from 'typeorm';
-import { CreatePostDto } from './dto/post-create.dto';
+import { In, Repository } from 'typeorm';
 import { UpdatePostDto } from './dto/post-update.dto';
 import { ViewPostDto } from './dto/post-view.dto';
+import { jwtPayload } from '../auth/strategies/accesstoken.strategies';
+import { PostCategory } from 'src/entity/post-category.entity';
+import { PostCategoryDto } from './dto/post-category.dto';
+import { ConfigService } from '@nestjs/config';
+import { S3Utils } from './util/s3.util';
+import { FormPostDto } from './dto/post-create.dto';
+import { ImagesToS3 } from './dto/post-image.dto';
 
 @Injectable()
 export class PostService {
   constructor(
     @InjectRepository(Posts) private readonly postRepository: Repository<Posts>,
+    @InjectRepository(PostCategory)
+    readonly postCategoryRepository: Repository<PostCategory>,
+    private readonly configService: ConfigService,
   ) {}
 
+  async attachCategoryToPosts(cid: number[], postid: number, userid: number) {
+    const deleteCategory = await this.postCategoryRepository.query(
+      `DELETE FROM "post-category" WHERE "poid" = $1`,
```

Comment:
Đổi sang cách viết sau:
const queryOrganization = this.organizationRepository.createQueryBuilder();
    queryOrganization.innerJoin('Organization.members', 'members');
    queryOrganization.andWhere('members.user_id = :memberId', {
      memberId: `${memberId}`,
    });